Islandora/islandora_solr_search
# 7.x islandora 1872 - Allow custom value separator for Solr

**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-1872

* Other Relevant Links (Google Groups discussion, related pull requests, Release pull requests, etc.)

# What does this Pull Request do?

This will allow the user to specify a separator between multivalued fields in the search results.

# What's new?

* added new textfield in the admin form 'islandora_solr_search_field_value_separator'.
* added a check for empty values in the admin form validation handler. This is to prevent users from saving the form and having empty values passed to the display, causing results to look messy.

# How should this be tested?

* Ingest an object and create some multivalued fields (mods_name_personal_creator_namePart_ms for example).
* Navigate to your.site/admin/islandora/search/islandora_solr/settings and under the 'Default display settings' tab you will see a new field labelled 'Field value separator'.
* Enter test values, and also test that empty values are being set to NULL and falling back to default separators.
* Make sure your multivalued field is being displayed for search results.

# Interested parties
@Islandora/7-x-1-x-committers, @nhart
